drivers/sensor/tmag5273/tmag5273_trigger.c

```diff
@@ -0,0 +1,461 @@
+/*
+ * Copyright (c) 2023 deveritec GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_tmag5273
+
+#include "tmag5273.h"
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/sensor/tmag5273.h>
+#include <zephyr/sys/util.h>
+
+#include <zephyr/pm/device.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_DECLARE(TMAG5273, CONFIG_SENSOR_LOG_LEVEL);
+
+/**
+ * @brief	Activate/deactivate the input interrupt on SoC
+ *
+ * @param	gpio_int	 interrupt pin handle
+ * @param	enable		\c true to enable; \c false otherwise
+ *
+ * @return see @ref gpio_pin_interrupt_configure_dt
+ */
+static int soc_gpio_irq_config(const struct gpio_dt_spec *gpio_int, const bool enable)
+{
+	if (gpio_int->port == NULL) {
+		return -EFAULT;
+	}
+
+	const gpio_flags_t flags = (enable) ? GPIO_INT_EDGE_FALLING : GPIO_INT_DISABLE;
+
+	return gpio_pin_interrupt_configure_dt(gpio_int, flags);
+}
+
+/**
+ * @brief change the activation state of the interrupts
+ *
+ * @param dev handle to the device
+ * @param mask \c true to activate interrupt masking on the sensor (deactivates the interrupt),
+ *             \c false otherwise
+ *
+ * @retval \c 0 on success
+ * @retval \c -EIO on communication error
+ */
+static int tmag5273_trigger_mask_interrupt(const struct device *dev, const bool mask)
+{
+	const struct tmag5273_config *drv_cfg = dev->config;
+	struct tmag5273_data *drv_data = dev->data;
+
+	int retval = 0;
+
+	retval = soc_gpio_irq_config(&drv_data->int_gpio, !mask);
+
+	if (retval < 0) {
+		LOG_DBG("error changing interrupt config on SoC |%d|", retval);
+		return -EIO;
+	}
+
+	const uint8_t intb_mask = (mask) ?
+		TMAG5273_INT_MASK_INTB_MASKED : TMAG5273_INT_MASK_INTB_ENABLED;
+
+	retval = i2c_reg_update_byte_dt(&drv_cfg->i2c,
+									TMAG5273_REG_INT_CONFIG_1,
+									TMAG5273_INT_MASK_INTB_MSK,
+									(intb_mask << TMAG5273_INT_MASK_INTB_POS));
+
+	if (retval < 0) {
+		LOG_DBG("error changing masking on sensor |%d|", retval);
+		retval = -EIO;
+	}
+
+	return retval;
+}
+
+/**
+ * @brief calculates the magnetic threshold
+ *
+ * @param val threshold value to be set in mT
+ * @param range range of the magnetic measurement
+ *
+ * @return threshold in on-chip representation
+ */
+static inline uint8_t tmag5273_trigger_calc_magn_threshold(const struct sensor_value *val,
+						uint16_t range)
+{
+	if (val->val1 == 0) {
+		return 0;
+	}
+
+	const uint8_t base_val =
+		((range == TMAG5273_MEAS_RANGE_LOW_MT_VER1) || (range == TMAG5273_MEAS_RANGE_HIGH_MT_VER1))
+			? TMAG5273_MEAS_RANGE_LOW_MT_VER1 : TMAG5273_MEAS_RANGE_LOW_MT_VER2;
+
+	return (uint8_t)(base_val * (1 + range) / 128 * val->val1);
+}
+
+/* ----------------------------------------------------------- */
+
+int tmag5273_trigger_attr_set(const struct device *dev,
```

Comment:
```suggestion
static int tmag5273_trigger_attr_set(const struct device *dev,
```

Reply:
This is currently called from tmag5273.c and thus part of the header. Unless I change the structure this won't be applicable.